pgoapi/pgoapi.py

```diff
@@ -623,39 +639,65 @@ def cleanup_pokemon(self, inventory_items=None):
         caught_pokemon = self.get_caught_pokemons(inventory_items)
         for pokemons in caught_pokemon.values():
             if len(pokemons) > self.MIN_SIMILAR_POKEMON:
-                # highest lvl pokemon first
-                sorted_pokemons = sorted(pokemons, key=self.pokemon_lvl, reverse=True)
-                for pokemon in sorted_pokemons[self.MIN_SIMILAR_POKEMON:]:
-                    if self.is_pokemon_eligible_for_transfer(pokemon, sorted_pokemons[0]):
+                # highest scoring pokemon first
+                sorted_pokemons = sorted(pokemons, key=self.pokemon_score, reverse=True)
+                pokemons_keep_counter = 0
+                pokemons_keep_iv_counter = 0
+                for pokemon in sorted_pokemons:
+                    eligible, keep_iv = self.is_pokemon_eligible_for_transfer(pokemon, sorted_pokemons[0],
+                                                                              pokemons_keep_counter,
+                                                                              pokemons_keep_iv_counter)
+                    if (eligible and pokemons_keep_counter > self.MIN_SIMILAR_POKEMON)\
+                            or pokemons_keep_counter >= self.MAX_SIMILAR_POKEMON:
                         self.do_release_pokemon(pokemon)
+                        continue
+                    else:
+                        pokemons_keep_counter += 1
+                        if keep_iv:
+                            pokemons_keep_iv_counter += 1
 
-    def is_pokemon_eligible_for_transfer(self, pokemon, best_pokemon):
+    def is_pokemon_eligible_for_transfer(self, pokemon, best_pokemon=Pokemon(),
+                                         pokemons_keep_counter=0, pokemons_keep_iv_counter=0):
         # never release favorites and other defined pokemons
         if pokemon.is_favorite or pokemon.pokemon_id in self.keep_pokemon_ids:
-            return False
-        elif self.RELEASE_DUPLICATES and (
-                    self.pokemon_lvl(best_pokemon) * self.RELEASE_DUPLICATES_SCALER > self.pokemon_lvl(
-                    pokemon) and self.pokemon_lvl(pokemon) < self.RELEASE_DUPLICATES_MAX_LV):
-            return True
-        # release defined throwaway pokemons  but make sure we have kept at least 1 (dont throw away all of them)
-        elif pokemon.pokemon_id in self.throw_pokemon_ids:
-            return True
+            return False, False
```

Comment:
I wanted to see how do we maintain a max of KEEP_POKEMON_ID pokemons, if i hit a hotspot containing pikachus like i did in japan i got 45 pikachus by morning, it could have filled up my bag. so the Keep_pokemon_id should also respect MAX_SIMILAR_POKEMON so if its favorite can be handled separately to return false false
ifs keep pokemon and the keep_pokemon count is >= max_similar pokemon, throw it away. or keep a different counter for MAX_FAVORITE_POKEMON set default to 9999